mongo.go

```diff
@@ -40,12 +42,21 @@ func newMongoItem(i *resource.Item) *mongoItem {
 func newItem(i *mongoItem) *resource.Item {
 	// Add the id back (we use the same map hoping the mongoItem won't be stored back)
 	i.Payload["id"] = i.ID
-	return &resource.Item{
+	item := &resource.Item{
 		ID:      i.ID,
 		ETag:    i.ETag,
 		Updated: i.Updated,
 		Payload: i.Payload,
 	}
+
+	if item.ETag == "" {
+		if v, ok := i.ID.(bson.ObjectId); ok {
+			item.ETag = "p-" + v.Hex()
```

Comment:
This one should not be prefixed with `p-`, should it?

Reply:
I think it should, because we are using the raw ID value here, and not the MD5 hash.
The `if/else` segment just decides how to serialize the ID, but it is still DB ID.

Reply:
@rs, the `p-` symbolise a provisional query-time-generated id, and is used on L138 and L172 to determine which query to use when matching an Etag.

Reply:
Ok nm I read too fast.